src/kernel/arch/x86_64/gdt/gdt.h

```diff
@@ -0,0 +1,23 @@
+// A purpose for GDT
+
+#pragma once
+
+// Entry of table
+
+struct gdt_entry
```

Comment:
You should do a typedef for this struct.

Also this GDT doesn't seems right, can you double check the documentation from [osdev](https://wiki.osdev.org/GDT_Tutorial)

Reply:
I would also recommend reading of [Intel manual Vol. 3A](https://www.intel.com/content/dam/support/us/en/documents/processors/pentium4/sb/25366821.pdf) which explain a lot about GDT.